tests/hierarchy_driver.cc

```diff
@@ -80,38 +163,198 @@ int main(int argc, char *argv[])
 {
   namespace boost_po = boost::program_options;
 
-  using f_type = std::vector<std::string>;
-
   MPI_Init(&argc, &argv);
 
   boost_po::options_description cmd("Available options");
-  cmd.add_options()("filename,f", boost_po::value<f_type>()->multitoken(),
-                    "file(s) containing input parameters");
+  cmd.add_options()("help,h", "produce help message");
+  cmd.add_options()("filename,f", boost_po::value<std::string>()->multitoken(),
+                    "file containing input parameters");
   cmd.add_options()("dim,d", boost_po::value<int>(), "dimension");
+  cmd.add_options()("matrix_free,m", boost_po::value<bool>(),
+                    "use matrix-free algorithm");
 
   boost_po::variables_map vm;
   boost_po::store(boost_po::parse_command_line(argc, argv, cmd), vm);
   boost_po::notify(vm);
 
-  f_type filenames = {"hierarchy_input.info"};
+  if (vm.count("help"))
+  {
+    std::cout << cmd << std::endl;
```

Comment:
Do we have some kind of avoid `std::endl` policy?

Reply:
No, we don't